src/api/wayland/keyboard.rs

```diff
@@ -37,8 +38,19 @@ impl wayland_kbd::Handler for KbdHandler {
             };
             let vkcode = key_to_vkey(rawkey, keysym);
             let mut guard = eviter.lock().unwrap();
-	    // TODO implement ModifiersState
-            guard.push_back(Event::KeyboardInput(state, rawkey as u8, vkcode, ModifiersState::default()));
+            guard.push_back(
+                Event::KeyboardInput(
+                    state,
+                    rawkey as u8,
+                    vkcode,
+                    ModifiersState {
+                        shift: mods.shift ^ mods.caps_lock,
```

Comment:
I just want confirmation about this part.

Wayland-kbd exposes the shift and caps lock key independently, and the logical equivalent of the shift modifier is represented by this XOR. Is it the proper abstraction level?